base.lua

```diff
@@ -1322,79 +1356,146 @@ function courseplay:setWaypointIndex(vehicle, number,isRecording)
 end;
 
 function courseplay:getIsCourseplayDriving()
-	return self.cp.isDriving;
+	return self.cp.isDriving or false
 end;
 
 function courseplay:setIsCourseplayDriving(active)
-	--self:setCpVar('isDriving',active,courseplay.isClient)
 	self.cp.isDriving = active
 end;
 
---This is a copy from the Autodrive code "https://github.com/Stephan-S/FS19_AutoDrive" 
---all credits go to their Dev team
+--the same code as giants AIVehicle:startAIVehicle(helperIndex, noEventSend, startedFarmId), but customized for cp
+
 --All the code that has to be run on Server and Client from the "start_stop" file has to get in here
-function courseplay:onStartCpAIDriver()
-	self.forceIsActive = true
-    self.spec_motorized.stopMotorOnLeave = false
-    self.spec_enterable.disableCharacterOnLeave = false
-    self.spec_aiVehicle.isActive = true
-    self.steeringEnabled = false
-
-    if self.currentHelper == nil then
-		self.currentHelper = g_helperManager:getRandomHelper()
+function courseplay:onStartCpAIDriver(helperIndex,noEventSend, startedFarmId)
+	local spec = self.spec_aiVehicle
+    if not self:getIsAIActive() then
+        --giants code from AIVehicle:startAIVehicle()
+
+		if helperIndex ~= nil then
+            spec.currentHelper = g_helperManager:getHelperByIndex(helperIndex)
+        else
+            spec.currentHelper = g_helperManager:getRandomHelper()
+        end
+        g_helperManager:useHelper(spec.currentHelper)
+        spec.startedFarmId = startedFarmId
+        if g_server ~= nil then
+            g_farmManager:updateFarmStats(startedFarmId, "workersHired", 1)
+        end
+        if noEventSend == nil or noEventSend == false then
+            local event = AIVehicleSetStartedEventCP:new(self, nil, true, spec.currentHelper, startedFarmId)
+            if g_server ~= nil then
+                g_server:broadcastEvent(event, nil, nil, self)
+            else
+                g_client:getServerConnection():sendEvent(event)
+            end
+        end
+        AIVehicle.numHirablesHired = AIVehicle.numHirablesHired + 1
+        AIVehicle.hiredHirables[self] = self
         if self.setRandomVehicleCharacter ~= nil then
             self:setRandomVehicleCharacter()
-            self.cp.vehicleCharacter = self.spec_enterable.vehicleCharacter
         end
-        if self.spec_enterable.controllerFarmId ~= 0 then
-            self.spec_aiVehicle.startedFarmId = self.spec_enterable.controllerFarmId
+		spec.mapAIHotspot = ShowMapHotspotSetting.createMapHotSpot(self)
```

Comment:
So this would look something like spec.mapAIHotspot = CpMapHotSpot(globalSettings.showMapHotSpotSetting)